core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala

```diff
@@ -0,0 +1,140 @@
+
+import org.apache.spark.SparkException
+import org.apache.spark.storage.{BlockId, RDDBlockId}
+
+/**
+ * Object for grouping error messages from (most) exceptions thrown during query execution.
+ */
+private[spark] object SparkCoreErrors {
+  def rddBlockNotFoundError(blockId: BlockId, id: Int): Throwable = {
+    new Exception(s"Could not compute split, block $blockId of RDD $id not found")
+  }
+
+  def blockHaveBeenRemovedError(string: String): Throwable = {
+    new SparkException(s"Attempted to use $string after its blocks have been removed!")
+  }
+
+  def histogramOnEmptyRDDOrContainingInfinityOrNaNError(): Throwable = {
+    new UnsupportedOperationException(
+      "Histogram on either an empty RDD or RDD containing +/-infinity or NaN")
+  }
+
+  def emptyRDDError(): Throwable = {
+    new UnsupportedOperationException("empty RDD")
+  }
+
+  def pathNotSupportedError(path: String): Throwable = {
+    new IOException(s"Path: ${path} is a directory, which is not supported by the " +
+      s"record reader when `mapreduce.input.fileinputformat.input.dir.recursive` is false.")
+  }
+
+  def checkpointRDDBlockIdNotFoundError(rddBlockId: RDDBlockId): Throwable = {
+    new SparkException(s"Checkpoint block $rddBlockId not found! Either the executor " +
+      s"that originally checkpointed this partition is no longer alive, or the original RDD is " +
+      s"unpersisted. If this problem persists, you may consider using `rdd.checkpoint()` " +
+      s"instead, which is slower than local checkpointing but more fault-tolerant.")
+  }
+
+  def endOfStreamError(): Throwable = {
+    new java.util.NoSuchElementException("End of stream")
+  }
+
+  def cannotUseMapSideCombiningWithArrayKeyError(): Throwable = {
+    new SparkException("Cannot use map-side combining with array keys.")
+  }
+
+  def hashPartitionerCannotPartitionArrayKeyError(): Throwable = {
+    new SparkException("HashPartitioner cannot partition array keys.")
+  }
+
+  def reduceByKeyLocallyNotSupportArrayKeysError(): Throwable = {
+    new SparkException("reduceByKeyLocally() does not support array keys")
+  }
+
+  def noSuchElementException(): Throwable = {
+    new NoSuchElementException()
+  }
+
+  def rddLacksSparkContextError(): Throwable = {
+    new SparkException("This RDD lacks a SparkContext. It could happen in the following cases: " +
+      "\n(1) RDD transformations and actions are NOT invoked by the driver, but inside of other " +
+      "transformations; for example, rdd1.map(x => rdd2.values.count() * x) is invalid " +
+      "because the values transformation and count action cannot be performed inside of the " +
+      "rdd1.map transformation. For more information, see SPARK-5063.\n(2) When a Spark " +
+      "Streaming job recovers from checkpoint, this exception will be hit if a reference to " +
+      "an RDD not defined by the streaming job is used in DStream operations. For more " +
+      "information, See SPARK-13758.")
```

Comment:
ditto

Reply:
there are "\n" in the message so this couldn't use replaceAll("\n", " ")

Reply:
I see.
You can change to
|"rdd1.map transformation. For more information, see SPARK-5063.
|(2) When a Spark ...

Reply:
but if we do that, the 2nd line will access limit of 100 characters, right?

Reply:
can we split paragraph like """ ... """.stripMargin.replaceAll("\n", " ") + "\n" + """ ... """.stripMargin.replaceAll("\n", " ")

Reply:
> but if we do that, the 2nd line will access limit of 100 characters, right?

You can adjust the code and break line.

Reply:
can you write your solution down, I'm not fully understand it. if we break line, should we need replaceAll("\n", " "), which will also replace "\n" between 2 paragraphs

Reply:
> can we split paragraph like """ ... """.stripMargin.replaceAll("\n", " ") + "\n" + """ ... """.stripMargin.replaceAll("\n", " ")

Seems good!